tsl/src/nodes/gapfill/exec.c

```diff
@@ -374,6 +374,99 @@ get_boundary_expr_value(GapFillState *state, GapFillBoundary boundary, Expr *exp
 	return gapfill_datum_get_internal(arg_value, state->gapfill_typid);
 }
 
+typedef struct collect_boundary_context
+{
+	List *quals;
+	Var *ts_var;
+} collect_boundary_context;
```

Comment:
nit: I guess you dont' have to keep the same name for type and struct, probably better to omit struct name in that case completely.